stdlib/random.ml

```diff
@@ -163,22 +163,30 @@ let default = {
   State.idx = 0;
 }
 
-let bits () = State.bits default
-let int bound = State.int default bound
-let int32 bound = State.int32 default bound
-let nativeint bound = State.nativeint default bound
-let int64 bound = State.int64 default bound
-let float scale = State.float default scale
-let bool () = State.bool default
+let _ = CamlinternalDomain.(register_initialiser (fun st ->
+  st.random <- Obj.repr (mk_default ())))
 
-let full_init seed = State.full_init default seed
-let init seed = State.full_init default [| seed |]
+let current_state () =
+  let open CamlinternalDomain in
+  let st = get_dls_state () in
+  Obj.magic st.random
+
+let bits () = State.bits (current_state ())
+let int bound = State.int (current_state ()) bound
+let int32 bound = State.int32 (current_state ()) bound
+let nativeint bound = State.nativeint (current_state ()) bound
+let int64 bound = State.int64 (current_state ()) bound
+let float scale = State.float (current_state ()) scale
+let bool () = State.bool (current_state ())
+
+let full_init seed = State.full_init (current_state ()) seed
```

Comment:
We should update the mli documentation to highlight that `full_init`, `init` and `self_init` update just the current domain's seed value.